docs/design/datacontracts/StackWalk.md

```diff
@@ -0,0 +1,67 @@
+# Contract StackWalk
+
+This contract encapsulates support for StackWalking managed threads.
+
+## APIs of contract
+
+```csharp
+public interface IStackWalkHandle { };
+public interface IStackDataFrameHandle { };
+```
+
+```csharp
+// Creates a stack walk and returns a handle
+IStackWalkHandle CreateStackWalk(ThreadData threadData);
+// Iterates the stackWalkHandle to the next frame. If successful, returns true. Otherwise false.
+bool Next(IStackWalkHandle stackWalkHandle);
+// Gets the current frame from a stack walk and returns a IStackDataFrameHandle to it.
+IStackDataFrameHandle GetCurrentFrame(IStackWalkHandle stackWalkHandle);
```

Comment:
Instead of defining a handle interface to drive enumeration and nothing else, could we instead provide the following method?

```csharp
IEnumerable<IStackDataFrameHandle> EnumerateFrames(ThreadData threadData);
```

This method could lazily walk the frames (like how I implemented reading stress log messages in my StressLog cdac PR), which would preserve existing performance characteristics.